### PR TITLE
Simplified WelcomePage

### PR DESCRIPTION
**Pull request by tsareg**
All logic was inside AbstractWelcomePageContainer, while WelcomePage class itself did almost nothing except of rendering WelcomePageContainer.
So I've merged WelcomePage and WelcomePageContainer classes together.

### CODE REVIEW

**Comment by legastero** [comment]
looks good to me

File: features/welcome/components/AbstractWelcomePage.js [hunk 1]
```diff
@@ -17,37 +22,42 @@ export class AbstractWelcomePageContainer extends Component {
...
      * @protected
      * @returns {void}
      */
-    _onRoomNameChange(value) {
-        this.setState({ roomName: value });
+    _onJoinPress() {
+        this.props.dispatch(navigate({
+            screen: APP_SCREEN.CONFERENCE,
+            room: this.state.roomName,
+            navigator: this.props.navigator
```
Comment (lyubomir) [inline features/welcome/components/AbstractWelcomePage.js:39]: Our coding style requires sorting by alphabetical order!

File: features/welcome/components/AbstractWelcomePage.js [hunk 2]
```diff
@@ -17,37 +22,42 @@ export class AbstractWelcomePageContainer extends Component {
...
-    _onJoinPress() {
-        this.props.onJoin(this.state.roomName);
+    _onRoomNameChange(value) {
+        this.setState({ roomName: value });
     }
 }
 
 /**
- * WelcomePageContainer component's property types.
+ * WelcomePage component's property types.
```
Comment (lyubomir) [inline features/welcome/components/AbstractWelcomePage.js:56]: Well, AbstractWelcomePage is technically more accurate than WelcomePage. I know that given that WelcomePage is going to reuse them, one may claim that these are WelcomePage's property types. But there's no way for AbstractWelcomePage to know that.

File: features/welcome/components/WelcomePage.native.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
 import React from 'react';
```
Comment (lyubomir) [inline features/welcome/components/WelcomePage.native.js:1]: Why is this import necessary?

File: features/welcome/components/WelcomePage.native.js [hunk 4]
```diff
@@ -39,4 +40,6 @@ export class WelcomePageContainer extends AbstractWelcomePageContainer {
     }
 }
 
-WelcomePageContainer.propTypes = AbstractWelcomePageContainer.propTypes;
+WelcomePage.propTypes = AbstractWelcomePage.propTypes;
```
Comment (lyubomir) [inline features/welcome/components/WelcomePage.native.js:43]: Our coding style requires jsdocs on everything!

File: features/welcome/components/WelcomePage.web.js [hunk 5]
```diff
@@ -32,4 +33,6 @@ export class WelcomePageContainer extends AbstractWelcomePageContainer {
     }
 }
 
-WelcomePageContainer.propTypes = AbstractWelcomePageContainer.propTypes;
+WelcomePage.propTypes = AbstractWelcomePage.propTypes;
```
Comment (lyubomir) [inline features/welcome/components/WelcomePage.web.js:36]: Our coding style requires jsdocs on everything!

File: features/welcome/components/WelcomePage.web.js [hunk 6]
```diff
@@ -1,14 +1,15 @@
 import React from 'react';
```
Comment (lyubomir) [inline features/welcome/components/WelcomePage.web.js:1]: Why is this import necessary?

File: features/welcome/components/WelcomePage.native.js (hunk 3 above)
Comment (lyubomir) [inline features/welcome/components/WelcomePage.native.js:1]: As eslint puts it:

error  'React' must be in scope when using JSX  react/react-in-jsx-scope

File: features/welcome/components/WelcomePage.web.js (hunk 6 above)
Comment (lyubomir) [inline features/welcome/components/WelcomePage.web.js:1]: As eslint puts it:

error  'React' must be in scope when using JSX  react/react-in-jsx-scope

Comment (tsareg) [inline features/welcome/components/WelcomePage.web.js:1]: This is required for JSX markup to be converted into React element

**Comment by lyubomir** [comment]
@legastero & @tsareg, this is causing the following error for me upon clicking the JOIN button on Web:

AbstractWelcomePage.js:37 Uncaught TypeError: this.props.dispatch is not a function

**Comment by tsareg** [comment]
@lyubomir I'm not getting the error both on web and mobile. Do you still have it?

**Comment by lyubomir** [comment]
@tsareg, I cannot reproduce it reliably. Anyway, I didn't see how it could be caused by your PR so I merged it after testing on both Web & mobile. Thank you for the quick response & for the idea of this PR!
